stream-chat-android-client/src/main/java/io/getstream/chat/android/client/controllers/ChannelController.kt

```diff
@@ -119,11 +120,26 @@ public interface ChannelController {
     )
     public fun events(): ChatObservable
 
-    public fun subscribe(listener: (event: ChatEvent) -> Unit): Disposable
+    public fun subscribe(listener: ChatEventListener<ChatEvent>): Disposable
+
+    @Deprecated(
+        message = "Use subscribeFor with ChatEventListener parameter",
+        level = DeprecationLevel.WARNING
+    )
```

Comment:
This kind of Kotlin syntax will always resolve to the lambda parameter functions first (instead of the interface parameter functions), which means that all Kotlin users of these APIs will see warnings on their code, despite it being correct.

We can avoid this by adding this annotation to these deprecated functions, removing them from Kotlin call resolution entirely:

```kotlin
@SinceKotlin("99999.9")
```

Reply:
Interesting, I'm not seeing that behavior in the updated docs, but I'll add the annotation anyway just to be safe.